src/libraries/System.Private.CoreLib/src/System/Buffers/Text/FormattingHelpers.CountDigits.cs

```diff
@@ -109,36 +111,52 @@ public static int CountDigits(ulong value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int CountDigits(uint value)
```

Comment:
Update `CountDigits(ulong value)` above to take advantage of this?

Reply:
> Update `CountDigits(ulong value)` above to take advantage of this?

FYI: There is a 64-bit version in the comments of Lemire's post

Reply:
> FYI: There is a 64-bit version in the comments of Lemire's post

Which specifically? The variants I'd seen were only good up to a smaller number of bits, like 57.

Reply:
I assume Jan meant
```csharp
if (longValue <= uint.MaxValue)
    return CountDigits(unchecked((uint)longValue));
```
?

Reply:
More like:
```
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public static int CountDigits(ulong value)
        {
            int digits;
            uint part;
            if (value >= 1_000_000_000)
            {
                if (part >= 1_000_000_000_000_000_000)
                {
                    part = (uint)(value / 1_000_000_000_000_000_000);
                    digits = 18;
                }
                else
                {
                    part = (uint)(value / 1_000_000_000);
                    digits = 9;
                }
            }
            else
            {
                part = (uint)value;
                digits = 0;
            }
            
            return digits + CountDigits(part);
        }
```

Reply:
> More like

Yeah, that's what I was planning on doing, but I want to experiment.